great_expectations/rule_based_profiler/data_assistant/data_assistant.py

```diff
@@ -0,0 +1,297 @@
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List, Optional, Union
+
+from great_expectations.core import ExpectationConfiguration, ExpectationSuite
+from great_expectations.core.batch import BatchRequestBase
+from great_expectations.data_context import BaseDataContext
+from great_expectations.execution_engine.execution_engine import MetricDomainTypes
+from great_expectations.rule_based_profiler.domain_builder import DomainBuilder
+from great_expectations.rule_based_profiler.expectation_configuration_builder import (
+    ExpectationConfigurationBuilder,
+)
+from great_expectations.rule_based_profiler.helpers.util import (
+    convert_variables_to_dict,
+)
+from great_expectations.rule_based_profiler.helpers.util import (
+    get_validator as get_validator_using_batch_list_or_batch_request,
+)
+from great_expectations.rule_based_profiler.parameter_builder import ParameterBuilder
+from great_expectations.rule_based_profiler.rule import Rule
+from great_expectations.rule_based_profiler.rule_based_profiler import (
+    BaseRuleBasedProfiler,
+    RuleBasedProfiler,
+)
+from great_expectations.rule_based_profiler.types import (
+    FULLY_QUALIFIED_PARAMETER_NAME_METADATA_KEY,
+    FULLY_QUALIFIED_PARAMETER_NAME_SEPARATOR_CHARACTER,
+    FULLY_QUALIFIED_PARAMETER_NAME_VALUE_KEY,
+    Domain,
+)
+
+
+class DataAssistant(ABC):
+    """
+    DataAssistant is an application built on top of the Rule-Based Profiler component.
+    DataAssistant subclasses provide exploration and validation of particular aspects of specified data Batch objects.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        batch_request: Union[BatchRequestBase, dict],
+        data_context: BaseDataContext = None,
+    ):
+        """
+        DataAssistant subclasses build "RuleBasedProfiler" to contain Rule configurations to embody profiling behaviors,
+        corresponding to indended exploration and validation goals.  Then executing "RuleBasedProfiler.run()" yields
+        "RuleBasedProfilerResult" object, containing metrics by "Domain", list of "ExpectationConfiguration" objects,
+        and overall "ExpectationSuite" object, immediately available for validating underlying data "Batch" objects.
+
```

Comment:
What would be the primary method in which a user would interact with the DataAssistant? Would it be a series of function calls? Would we be able to include that in this description here?

Reply:
@Shinnnyshinshin Good point -- it is seen in a test, but I will supply it here at the top (consider it done!).  Thanks!

Reply:
@Shinnnyshinshin Committed.  Thanks!